### ISSUE TITLE
Developers Discussion 

### ISSUE THREAD

**Issue by LinxGem33** [LinxGem33/Neon#1]

**Comment by Elbullazul** [comment]
@LinxGem33 can you explain what's your idea for this project?

**Comment by LinxGem33** [comment]
@Elbullazul 

i wanted to discuss some ideas with you what theme to base this off adapta or numix i prefer adapta also i have some art work which i started just min close and max buttons but im currently working on the rest.
ill create a folder named artwork for the artwork so you can get some idea and we can take it from there.

what do you think?

**Comment by Elbullazul** [comment]
ok, let's go with adapta. What's the general appearance you want for the theme? Flat, low-contrast borders, animations, bright color touches? a bit of roundness? 

the ideal would be an image of what you want, so we can sync our ideas on theme creation

**Comment by LinxGem33** [comment]
@Elbullazul 

what ill do is draw some mock ups and push them to the artwork folder but we will defiantly have to reduce the header bar for adapta i already have alternative shell themes i can work on once we have agreed on the direction and art work etc we can get started :)

kind regards

**Comment by Elbullazul** [comment]
@LinxGem33 any news?

**Comment by LinxGem33** [comment]
@Elbullazul 

hey its been a busy couple of weeks I’ve got to update a couple of my projects but should have the mock-ups with you shortly, are you ready to get started? you need to download the latest adapta theme 3.20 and on wards I’ve also been working on the shell theme.

kind regards

**Comment by Elbullazul** [comment]
@LinxGem33 allright, I'll get the source

**Comment by LinxGem33** [comment]
@Elbullazul 

I’ve uploaded some wire frame mock-ups you should get an idea what direction we are going we can change all colours etc when we get started if you want to get started with the header bar and buttons ill start on the gnome shell let me know what ideas you have etc?

kind regards

**Comment by paullinuxthemer** [comment]
Hello,

I'm new here.  
For the sake of Github, I've named myself paullinuxthemer.  But you might know me as Paulxfce, or Manjarrocinnamonfan or even Paulubuntu.  Al aliases that I've created during the 10 years I've been on gnomelook.org. At LinxGem33 request I join this channel.  

I may know something about theming, but nothing about Github and it's purpose.  Upon till now, I've always done everything the 'old fashion'-way  (which I still prefer). But if this is to be a collaborating thingy, I guess this github-way-of-working is necessary.   So anyway, a little help on this would be appreciated.   

If my english is a bit awkward, it is because I'm dutch-speaking (I live in Belgium, 30 miles from Brussels).

**Comment by LinxGem33** [comment]
@paullinuxthemer 

welcome, just to give you a idea of how to add code or change it you can edit code on any of the specific themes your working on be it from github direct or on your local machine to use github direct you need to follow these steps first you need to send what’s called a pull request how you do that is as follows;

1.click on pull request located at the top next to code & issue
2.click new pull request
3.choose master branch once all changes are made etc
4.once request is pushed it will then be checked for coding errors and proofing before being committed
5.your changes are excepted

this is just one way of pushing code to the repo, I’ve left this example of a contributor pushing code to the OSX-Arc-White repo below to give you an idea.

Click > [Example of a pull request](https://github.com/LinxGem33/OSX-Arc-White/pull/12)

please don’t hesitate to ask questions if your still unsure.

kind regards

**Comment by paullinuxthemer** [comment]
So, about this Neon-project: 

Some questions:

"Neon" as in neon-lights? Or "neon" as in 'new'? 
Any color-pallet in mind?
Blocky or rounded? Dark or light theme?
For which Desktop Environment? 
Only gtk 3.20 +?

I've seen that you plan to use the Adapta-theme as a starting point?  I ask, because I've briefly helped one of the contributors on the Adapta-theme.  And I have to say that he was the first person I've encountered which actually knew what he was doing.  I attempted to mod his theme too, only to find out that it is a whole different ballgame.  It is nothing like any other theme (in fact way beyond my comprehension, I admit) I too tried to change the titlebuttons on this theme.  But I didn't manage. So I'm not going to be much help here.

**Comment by LinxGem33** [comment]
@paullinuxthemer 

left you some information above , don’t worry about the adapta theme we are only using the frames not the header bars Ive worked on the gnome shell for a little while now and the header bars @Elbullazul is taking a look at them but if you look in the art work folder ive done some rough moc ups using adapta frames with arc header bars as adapta is based on flat-plat we have a good broad base to work from.

1.as in neon lights
2.four different colours eventually
3.not sure yet leaning towards rounded
4.gnome to start then add more later
5.yes 3.20 and above

also my gnome shell theme for the osx series is from the adapta gnome shell took me a while to do but its defiantly workable i transformed it to look like the arc shell.

what ideas have you got as nothing is set in stone yet?

kind regards

**Comment by paullinuxthemer** [comment]
I lean towards the arc theme as a basis.  It is a solid theme. You can throw any application to it and it still looks good.  At the same time it is versatile. And it is an ideal theme to use if you want to expand your Neon-theme to other DE's.   My T4G-theme and the latest Gnome-OSX are mod's of the Arc theme. Yet they are very different.  I've spend a great deal of time on this theme.  So for me, it is easier to work with.  

I've no idea on how you make your changes but this is how I work:

I do not work with CSS, but with the SCSS-files in the SASS-file,  which translates into CSS.  

I do not change the png's as such. 
In the Arc theme for example, I generate the assets-file with the 'render-assets.sh' script, which translates the graphical elements you find in the 'assets.svg' into the png's .  So I do all my graphical work using SVG's with Inkscape.

**Comment by LinxGem33** [comment]
@paullinuxthemer 

personally i prefer the arc theme as it is easy to work with ill see if @Elbullazul  is comfortable working with the arc theme i dont think there will be a issue we can get started with getting this theme in motion the good news is me and you both have spent alot of time with the arc code base i just want something all developers feel comfortable using. so if your happy with that I am also.

I want to hopefully get started on some rough ideas and have a pre alpha release within a month or two once we are set on the direction what do you think?

kind regards

**Comment by paullinuxthemer** [comment]
I'll have Some buttons ready soon.

**Comment by paullinuxthemer** [comment]
Ok, here's a mock-up:
![mock-up-neon-buttons](https://cloud.githubusercontent.com/assets/26058111/23397921/b73e5a70-fd9a-11e6-9703-cef76b808e3b.png)

**Comment by paullinuxthemer** [comment]
Uploaded it to the Artwork/buttons file.

**Comment by Elbullazul** [comment]
@paullinuxthemer nice buttons!

about the arc theme, shure, I can get used to it

**Comment by paullinuxthemer** [comment]
OK ! 
What about the colors?  They are a bit flashy.  I'm concerned that neon-colors are a strain on the eyes, so I made them a bit less flashy. Still, I'm afraid that - even using this color-pallet- is still to much.

**Comment by LinxGem33** [comment]
@paullinuxthemer & @Elbullazul 

nice buttons, that’s settled then we will go with arc as a base what colours should we go with? I’m thinking three distinctive colours to start also are we going with flat or shadowed panel?

kind regards

**Comment by paullinuxthemer** [comment]
Also, I'm thinking of porting the assets from the Vimix theme to the Arc-assets.  And -in the proces: changing the colors.  So the buttons look a bit more like those on the Adapta theme, but less square
![g9664](https://cloud.githubusercontent.com/assets/26058111/23408800/b0337f10-fdc9-11e6-9b10-c6bbb29c3e76.png)
.

**Comment by paullinuxthemer** [comment]
The thing is.  Do we settle for a light theme (which I prefer, when using bright colors) or a darker version? Or both? 
A flat headerbar, or gradient? transparency or not? 
I agree with the colors, just a question of finding the right kind.

**Comment by LinxGem33** [comment]
@paullinuxthemer 
I’ve made some rough button mock up  colours for light and dark themes below

**dark button**
![button-colours](https://cloud.githubusercontent.com/assets/19158615/23409523/ce3d6f30-fdc3-11e6-999c-b14706686612.png)

**hover sate colour**
![button-colours-full](https://cloud.githubusercontent.com/assets/19158615/23409538/d84a4390-fdc3-11e6-90fd-6710f2ae6b0e.png)

**light button**
![button-colours-light](https://cloud.githubusercontent.com/assets/19158615/23409544/dd8e6a52-fdc3-11e6-8087-e29cca0a1cde.png)

definitely keep the ideas coming its getting me thinking and creating we can try flat and gradient just to see the difference but i think the colours should stand out but not be to powerful let me know what you think of the button colours.

kind regards

**Comment by Elbullazul** [comment]
colors are ok but too flashy, maybe add a little alpha?

**Comment by Elbullazul** [comment]
@LinxGem33 also, may I suggest we use the same colors for window buttons and the other neon colors instead be applied to selected bg color and selected borders and widgets

**Comment by paullinuxthemer** [comment]
Do you mean that all window-buttons have the same 'circle'-color? So 3 times the same button in their normal state?

**Comment by Elbullazul** [comment]
@paullinuxthemer no, but for each color variation of the theme that we keep the same window colors to unify the  overall appearance

**Comment by paullinuxthemer** [comment]
I'm a bit confused here: the button-colors (let's say pink-red for close; orange-yellow for minimize and some thing greenish for maximize) stay the same.  But the headerbar-colors (and such)  can vary.  
Or is it the other way around: button-colors can vary, headerbar-colors (and such) stay the same?

**Comment by LinxGem33** [comment]
@paullinuxthemer  & @Elbullazul 

yeah the colours are a bit strong its just a rough idea i think that if paullinuxthemer if you can work on the buttons for the min/max/close and the shell buttons and if you want to incorporate the  assets from the Vimix theme

![mock-up-neon-buttons](https://cloud.githubusercontent.com/assets/26058111/23397921/b73e5a70-fd9a-11e6-9703-cef76b808e3b.png)

![g9664](https://cloud.githubusercontent.com/assets/26058111/23408800/b0337f10-fdc9-11e6-9b10-c6bbb29c3e76.png)

and once youve finished the artwork ill work on incorporating the assets into the shell theme and gtk theme etc  once I’ve finished then Elbullazul if you can follow paullinuxthemer colour scheme for the button borders and window colours and then hopefully we will have something to work with and tweak

what do you guys think?

kind regards

**Comment by paullinuxthemer** [comment]
Currently trying to copy the vimix-'assets.svg'-template into the Arc-theme.  Which proves to more of a hassle than I anticipated.  The naming of the assets is different, some are missing and some are redundant.  And for some  I have to create new. There are about 132 items to check and correct. I've done about 90 of them so far.  This is only for GTK 2.0.  Then there is the assets.svg of the GTK 3.0 to do (84 elements to do). This will keep me busy for a few days...   Once the 'assets.svg' of both are finished, we will have the graphical elements ready.

**Comment by LinxGem33** [comment]
@paullinuxthemer 

brilliant once you’ve finished just push them to he repo and i can get started on incorporating them into the gtk theme and gnome shell let me know when you’ve finished and ill get started.

kind regards

**Comment by paullinuxthemer** [comment]
@LinxGem33 
This is a stupid way to try to get a hold on you...
Anyway, it may be a stupid question, but I can't seem to get 'bold'-text into the description of the themes we post on gnome-look.  I noticed you did.  Is it html-code?

**Comment by LinxGem33** [comment]
@paullinuxthemer 

its actually bb code, the code you need for bold text is below:

[b]>>insert your text here<<[/b]

kind regards

**Comment by paullinuxthemer** [comment]
Thanks @LinxGem33

**Comment by paullinuxthemer** [comment]
Working on this crazy idea: 

![scr9](https://cloud.githubusercontent.com/assets/26058111/26221905/5fbb3900-3c18-11e7-90ac-7fc7ad76d48b.jpg)


This is the working theme:
[Gn-Miui8.zip](https://github.com/LinxGem33/Neon/files/1012346/Gn-Miui8.zip)

**Comment by LinxGem33** [comment]
@paullinuxthemer 

really like the rounded buttons, if i was to suggest something it would be to follow a colour palate

![cpal](https://cloud.githubusercontent.com/assets/19158615/26222590/8794a9da-3c12-11e7-8bc4-d268f30977e7.png)
try and make your colours follow the same hex family it always makes the look feel so much more professional, but this theme really has a lot of potential.

are you using the flat-plat theme or adapta as a base?

**Comment by paullinuxthemer** [comment]
actually this is VIMIX.  But I changed allot.  I used images for the headerbar to create the landscape.  
What I like is the fact that it uses 4 different colors in the headerbar.  Blue for typical gnome-apps; dark-grey for windows with a menubar in it; and custom-theming nautilus (green-grey) and tweak-tool is also custome-themed (teal)

**Comment by LinxGem33** [comment]
@paullinuxthemer 

yeah definitely test and see how far you can take this theme, i think after i update my theme for gnome 3.24 I’m going to start developing my new theme i have a few ideas in place and I’m going to try concentrate solely on that project and develop it just for gnome.

**Comment by Elbullazul** [comment]
any news?

**Comment by LinxGem33** [comment]
@Elbullazul 
Ive been working on my new project [Arc menu](https://extensions.gnome.org/extension/1228/arc-menu/) which has been taking alot of my time what projects are you running at the moment? congrats by the way on your unity 7 shell theme.

**Comment by Elbullazul** [comment]
@LinxGem33 thanks

I'm adapting my themes to a modular structure to batch apply updates easily, and polishing 3.20 compatibility

**Comment by LinxGem33** [comment]
@Elbullazul 

how is setting up the theme collection coming along with the themes, let me know when your ready for me to start directing trafic towards OSX Arc Group for issues etc. 

and thank you again its appreciated.

kind regards

**Comment by Elbullazul** [comment]
@LinxGem33 it's a bit harder than what I thought. I have to re-define colors and that will take some time. But otherwise SASS looks like a great tool. Should've started using it a lot sooner

**Comment by Elbullazul** [comment]
oops, misread your question, sorry.

Arc group setup is almost complete, except for the repos, everything's OK. but like I said I'm a bit busy myself adapting to SASS, so probably no news until 1-2 weeks
